app/components/works/form_component.rb

```diff
@@ -29,5 +31,17 @@ def work
     def work_version
       work_form.model.fetch(:work_version)
     end
+
+    def doi_field
+      return "https://doi.org/#{Doi.for(druid)}." if doi
+
+      # create DOI URL or placeholder for works in collections that allow DOI assignment
+      collection = work.collection
+      return unless (collection.doi_option == 'depositor-selects' && work.assign_doi) || collection.doi_option == 'yes'
+
+      return "https://doi.org/#{Doi.for(druid)}." if druid
+
+      ':DOI will be inserted here automatically when available:'
```

Comment:
Should this use `WorkVersion::DOI_TEXT`?

Reply:
Right, that's better. Fixed.